lab4.py

```diff
@@ -0,0 +1,96 @@
+class FiniteAutomata:
+    def __init__(self, Q, q0, delta, F, sigma, dfa):
+        self.Q = Q #set of states
+        self.q0 = q0 #initial state
+        self.delta = delta #transitions
+        self.F = F #final state
+        self.sigma = sigma #terminals
+        self.dfa = dfa
+
+    @staticmethod
+    def removeWhiteSpace(stuff):
+        '''
+        strips the strings from a list of any white spaces
+        '''
+        for i in range(len(stuff)):
+            stuff[i] = stuff[i].strip()
+        return stuff
+
+    @staticmethod
+    def makeTransitions(line):
+        '''
+         algorithm used for delta lines, i.e. lines of the form "something,other = stuff"
+        '''
+        stuff = line.split("=")
+        lhs = FiniteAutomata.removeWhiteSpace(stuff[0].split(","))
+        rhs = FiniteAutomata.removeWhiteSpace(stuff[1].split(","))
+        return [lhs, rhs[0]]
+
+    @staticmethod
+    def readFromFile(filename = "D:\\Facultate\\Sem 1\\FLCD\\FA.in"):
+        '''
+        reading and parsing the FA from the file
+        '''
+        file = open(filename)
+
+        Q = FiniteAutomata.removeWhiteSpace(file.readline().split(","))
+        sigma = FiniteAutomata.removeWhiteSpace(file.readline().split(","))
+        q0 = FiniteAutomata.removeWhiteSpace(file.readline().split(","))
+        F = FiniteAutomata.removeWhiteSpace(file.readline().split(","))
+        delta = {}
+        line = file.readline()
+        dfa = True
+        while line:
+            transition = FiniteAutomata.makeTransitions(line)
+            dfa = True
```

Comment:
Do you use dfa?

Reply:
Yes, in the checkString function. Dfa is a bool value which tells the nature of the FA, that is, if the FA is a deterministic or not